homeassistant/components/anova/sensor.py

```diff
@@ -0,0 +1,106 @@
+    SensorEntityDescription(
+        key=AnovaPrecisionCookerSensor.COOK_TIME,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        native_unit_of_measurement=UnitOfTime.SECONDS,
+        icon="mdi:clock-outline",
+        name="Cook Time",
+    ),
+    SensorEntityDescription(key=AnovaPrecisionCookerSensor.STATE, name="State"),
+    SensorEntityDescription(key=AnovaPrecisionCookerSensor.MODE, name="Mode"),
+    SensorEntityDescription(
+        key=AnovaPrecisionCookerSensor.TARGET_TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:thermometer",
+        name="Target Temperature",
+    ),
+    SensorEntityDescription(
+        key=AnovaPrecisionCookerSensor.COOK_TIME_REMAINING,
+        native_unit_of_measurement=UnitOfTime.SECONDS,
+        icon="mdi:clock-outline",
+        name="Cook Time Remaining",
+    ),
+    SensorEntityDescription(
+        key=AnovaPrecisionCookerSensor.HEATER_TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:thermometer",
+        name="Heater Temperature",
+    ),
+    SensorEntityDescription(
+        key=AnovaPrecisionCookerSensor.TRIAC_TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:thermometer",
+        name="Triac Temperature",
+    ),
+    SensorEntityDescription(
+        key=AnovaPrecisionCookerSensor.WATER_TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:thermometer",
+        name="Water Temperature",
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: config_entries.ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Anova device."""
+    anova_data: AnovaData = hass.data[DOMAIN][entry.entry_id]
+    sensors: list[AnovaSensor] = []
+    for coordinator in anova_data.coordinators:
+        sensors.extend(
+            [
+                AnovaSensor(coordinator, description)
+                for description in SENSOR_DESCRIPTIONS
+            ]
+        )
+    async_add_entities(sensors)
+
+
+class AnovaSensor(AnovaEntity, SensorEntity):
+    """A sensor using anova coordinator."""
+
+    def __init__(
+        self, coordinator: AnovaCoordinator, description: SensorEntityDescription
+    ) -> None:
+        """Set up an Anova Sensor Entity."""
+        super().__init__(coordinator)
+        self.entity_description = description
+        self._attr_unique_id = f"{coordinator._device_unique_id}_{description.key}"
```

Comment:
If everything is going to use entity description you could make the base class take the description and move this all into the base class.

Reply:
My only thought is going forward - I plan to add a climate entity, which then wouldn't have a description key for the unique id. What are your thoughts about that @bdraco?

Reply:
If you're going to add other entity types that use descriptions you could add another base class for description entities on top of your existing base class.

If you aren't than I'd leave it as is

Reply:
I'll add binary sensors, so I created a description entity which inherits the regular anovaentity. And sensor and binary sensor will be description entities